### PR TITLE
#projects #refactor project.c: Changed Camel To Snake Case And Used Macros In is_valid_project_name()

### PR DESCRIPTION
**Pull request by RHL120**
**Checklist**

- [ ] Closing issues: #issue
- [x] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [radare2book](https://github.com/radareorg/radare2book)

**Description**
I tried to check if I renamed all camel case variables properly but you should propbaly double check weather or not I gave a variable a stupid name. I used the macros IS_UPPER and IS_LOWER in is_valid_project_name()
I made he paramter of remove_notes_file() and remove_rop_directory() constant.
The function r_core_project_save() now uses r_retrun_val_if_fail ().

### CODE REVIEW

File: libr/core/project.c [hunk 1]
```diff
@@ -21,10 +21,7 @@ static bool is_valid_project_name(const char *name) {
 		case '-':
 			continue;
 		}
-		if (name[i] >= 'a' && name[i] <= 'z') {
-			continue;
-		}
-		if (name[i] >= 'A' && name[i] <= 'Z') {
+		if (IS_LOWER(name[i]) || IS_UPPER(name[i])) {
```
Comment (trufae) [inline libr/core/project.c:24]: space before (

File: libr/core/project.c [hunk 2]
```diff
@@ -151,17 +148,17 @@ static inline void remove_project_file(char * path) {
 		}
 }
 
-static inline void remove_notes_file(char *prjDir) {
-		char *notes_txt = r_str_newf ("%s%s%s", prjDir, R_SYS_DIR, "notes.txt");
+static inline void remove_notes_file(char *pd) {
```
Comment (trufae) [inline libr/core/project.c:151]: const char *pd

File: libr/core/project.c [hunk 3]
```diff
@@ -151,17 +148,17 @@ static inline void remove_project_file(char * path) {
...
 		if (r_file_exists (notes_txt)) {
 			r_file_rm (notes_txt);
 			eprintf ("rm %s\n", notes_txt);
 		}
 		free(notes_txt);
 }
 
-static inline void remove_rop_directory(char *prjDir) {
-		char *rop_d = r_str_newf ("%s%s%s", prjDir, R_SYS_DIR, "rop.d");
+static inline void remove_rop_directory(char *prj_dir) {
```
Comment (trufae) [inline libr/core/project.c:160]: const char*

File: libr/core/project.c [hunk 4]
```diff
@@ -770,56 +767,56 @@ R_API bool r_core_project_save_script(RCore *core, const char *file, int opts) {
...
 	bool ret = true;
-	char *scriptPath, *prjDir;
+	char *script_path, *prj_dir;
 	SdbListIter *it;
 	SdbNs *ns;
-	char *oldPrjName = NULL;
+	char *old_prj_name = NULL;
 
-	if (!prjName || !*prjName) {
+	if (!prj_name || !*prj_name) {
```
Comment (trufae) [inline libr/core/project.c:778]: use r_return_val_if_fail instead

**Review by trufae — CHANGES_REQUESTED** [review]

File: libr/core/project.c [hunk 5]
```diff
@@ -774,7 +774,7 @@ R_API bool r_core_project_save(RCore *core, const char *prj_name) {
 	SdbListIter *it;
 	SdbNs *ns;
 	char *old_prj_name = NULL;
-	r_return_val_if_fail (false, !prj_name || !*prj_name);
+	r_return_val_if_fail (false, prj_name || *prj_name);
```
Comment (trufae) [inline libr/core/project.c:777]: Should be &&

File: libr/core/project.c [hunk 6]
```diff
@@ -21,10 +21,7 @@ static bool is_valid_project_name(const char *name) {
 		case '-':
 			continue;
 		}
-		if (name[i] >= 'a' && name[i] <= 'z') {
-			continue;
-		}
-		if (name[i] >= 'A' && name[i] <= 'Z') {
+		if (IS_LOWER (name[i]) || IS_UPPER (name[i])) {
```
Comment (trufae) [inline libr/core/project.c:24]: if (isalpha(name[i]))

File: libr/core/project.c [hunk 7]
```diff
@@ -345,38 +342,38 @@ R_API void r_core_project_execute_cmds(RCore *core, const char *prjfile) {
...
+	free (ps->prj_name);
+	free (ps->rc_path);
 	free (ps);
 	return R_TH_STOP;
 }
 
-R_API RThread *r_core_project_load_bg(RCore *core, const char *prjName, const char *rcPath) {
-	ProjectState *ps = R_NEW0 (ProjectState);
+R_API RThread *r_core_project_load_bg(RCore *core, const char *prj_name, const char *rc_path) {
+	project_state *ps = R_NEW0 (project_state);
```
Comment (trufae) [inline libr/core/project.c:359]: dont snakecase the types. ProjectState is fine

File: libr/core/project.c [hunk 8]
```diff
@@ -673,7 +670,7 @@ static bool project_save_script(RCore *core, const char *file, int opts) {
 				}
 			}
 		}
-		
+
```
Comment (trufae) [inline libr/core/project.c:673]: empty line

**Review by trufae — CHANGES_REQUESTED** [review]

**Comment by trufae** [comment]
Still red

**Comment by RHL120** [comment]
> Still red

Yeah I think I know why that happend. Sorry for being an idiot

**Comment by trufae** [comment]
Mistakes happen. You should try harder to become an idiot :) seems like this wasnt yet the reason for the ci to be red 🤔

**Pull request merged**